suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,312 @@
+        '''
+
+        # raise an error if this is the second `start` document seen.
+        if 'start' in self._meta['metadata']:
+            raise RuntimeError(
+                "The serializer in suitcase.tiff expects documents from one "
+                "run only. Two `start` documents where sent to it")
+
+        # add the start doc to self._meta and format self._file_prefix
+        self._meta['metadata']['start'] = doc
+        self._templated_file_prefix = self._file_prefix.format(**doc)
+
+        # return the start document
+        return doc
+
+
+    def stop(self, doc):
+        '''Add `stop` document information to the metadata dictionary.
+
+        This method adds the stop document information to the metadata
+        dictionary. In addition it also creates the metadata '.json' file and
+        exports the metadata dictionary to it.
+
+        Parameters:
+        -----------
+        doc : dict
+            The document dictionary associated with the stop message.
+        '''
+        # add the stop doc to self._meta.
+        self._meta['metadata']['stop'] = doc
+
+        # open a json file for the metadata and add self._meta to it.
+        f = self.manager.open('run_metadata',
+                              f'{self._templated_file_prefix}meta.json', 'xt')
+        json.dump(self._meta, f)
+        self.artifacts['run_metadata'].append(f)
+        self._files['meta'] = f
+
+        # return the stop document
+        return doc
+
+
+    def descriptor(self, doc):
+        '''Add `descriptor` document information to the metadata dictionary.
+
+        This method adds the descriptor document information to the metadata
+        dictionary. In addition it also creates the file for data with the
+        stream_name given by the descriptor doc for later use.
+
+        Parameters:
+        -----------
+        doc : dict
+            The document dictionary associated with the descriptor message.
+
+        '''
+        # extract some useful info from the doc
+        stream_name = doc.get('name')
+        filename = self._templated_file_prefix + f'{stream_name}.tiff'
+        # replace numpy objects with python ones to ensure json compatibility
+        sanitized_doc = event_model.sanitize_doc(doc)
+        # Add the doc to self._meta
+        self._meta['metadata']['descriptors'][stream_name] = sanitized_doc
+        # initialize some items in self._meta for use by event_page later
+        self._meta[stream_name]['seq_num'] = []
+        self._meta[stream_name]['time'] = []
+        self._meta[stream_name]['timestamps'] = {}
+        self._meta[stream_name]['uid'] = []
+        # open the file handle to write the event_page data to later
+        f = self.manager.open('stream_data', filename, 'xb')
+        # use the file handle to create the tiff file writing object
+        self._files[doc['uid']] = tifffile.TiffWriter(f, bigtiff=True)
+        self.artifacts['stream_data'].append(self._files[doc['uid']])
```

Comment:
I think this line is left over from the 'manager_artifacts' idea and should be removed.

Reply:
correct, it is now removed